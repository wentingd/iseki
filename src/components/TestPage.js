import React, { Component } from 'react';
import { withStyles } from '@material-ui/core/styles';
import Button from '@material-ui/core/Button';

const styles = theme => ({
  button: {
    margin: theme.spacing.unit,
  },
});


export class TestPage extends Component {
  constructor(props){
    super();
  }

    handleClick = () => {
      console.log('Hello from handleClick');
    }

    render() {
      const { classes } = this.props;
      return (
          <div>
            <Button color="primary" className={classes.button} onClick={this.handleClick()}>
              Primary
            </Button>
          </div>
      );
    }
}

export default withStyles(styles)(TestPage);
