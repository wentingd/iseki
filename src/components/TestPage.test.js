import React from 'react';
import TestPage from './TestPage';

it('should render the same', () => {
  const wrapper = shallow(<TestPage />);
  expect(wrapper).toMatchSnapshot();
});
