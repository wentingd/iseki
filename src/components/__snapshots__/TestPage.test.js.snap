// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the same 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(TestPage) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "classes": Object {
        "button": "TestPage-button-1",
        "input": "TestPage-input-2",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "classes": Object {
          "button": "TestPage-button-1",
          "input": "TestPage-input-2",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
